### ISSUE TITLE
digitalPinToInterrupt missing

### ISSUE THREAD

**Issue by xxxajk** [chipKIT32/chipKIT-core#114]
This portability function allows one to find out the interrupt number for a pin, or to find out if the pin is even capable of an interrupt at all by checking an array that maps from a pin to an interrupt number.
The idea is to allow dynamic IRQ assignment within a class, and more modern libraries use it.

Example use case:

```
        int intr = digitalPinToInterrupt(irq);
        if(intr == NOT_AN_INTERRUPT) {
                // error
                return -2;
        }
```

**Comment by xxxajk** [comment]
For UNO32, here is what can be used.
```
#define NOT_AN_INTERRUPT -1
#define digitalPinToInterrupt(p) ((p) == 2 ? 1 : ((p) == 7 ? 2 : ((p) == 8 ? 3 : ((p) == 35 ? 4 : ((p) == 38 ? 0 : NOT_AN_INTERRUPT)))))

```
What to do on other boards, I have no idea, since I do not own any other boards, but the macro will more-or-less be the same.

**Comment by xxxajk** [comment]
fixed in commit 2ab581b, closing...
